libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -733,21 +751,30 @@ instance (HasSpec fn a, Arbitrary (TypeSpec fn a)) => Arbitrary (Specification f
       ]
 
   shrink TrueSpec = []
-  shrink (MemberSpec xs) = TrueSpec : (MemberSpec . nub <$> shrinkList (shrinkWithSpec @fn TrueSpec) xs)
+  shrink (MemberSpec ys) = TrueSpec : mapMaybe mem (shrinkList (shrinkWithSpec @fn TrueSpec) (NE.toList ys))
+    where
+      mem xs =
+        case NE.nonEmpty (nub xs) of
+          Nothing -> Nothing
+          Just as -> Just $ MemberSpec as
   shrink (TypeSpec ts cant)
-    | null cant = TrueSpec : MemberSpec [] : map typeSpec (shrink ts) -- why MemberSpec[] ?
+    | null cant = TrueSpec : map typeSpec (shrink ts)
```

Comment:
Possibly you want to add back `ErrorSpec` here

Reply:
I did that, but I don't understand the reason why.